### PR TITLE
Feature/register frontend

### PR DESCRIPTION
**Pull request by jasonevangelista**
- register modal frontend
- input validation
- unit tests for modal inputs (except for e-mail input since may validation check for existing email sa db, di kaya sa unit test)

Let me know if may suggestions kayo sa design and colors! (I made the main color blue lang muna since it might be weird if red kasi yung red color sa modal nag iindicate if wrong input so baka magka confusion)

Screenshots of modal:

- Initial View of Modal

Screen Shot 2020-12-27 at 2 13 29 PM

- Sample Error Displays

Screen Shot 2020-12-27 at 2 20 58 PM

- Success Displays

Screen Shot 2020-12-27 at 2 14 01 PM

- Password Info Display

Screen Shot 2020-12-27 at 2 14 14 PM

### CODE REVIEW

**Comment by NeilMatthewLua** [comment]
Hi @jasonevangelista! Thank you for your continuous effort on this project! 

While testing on your branch, I've noticed that whenever I click on the Sign-Up button, the scroll bar on the side vanishes. (please see attachments below) Can you check it out on your end as well?

image
image

File: components/RegisterModal.js [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+                }
+                return Promise.resolve();
+              }
+            }
+          ]}>
+          <Input id="email" placeholder="e-mail" style={{ borderRadius: '7px' }} />
+        </Form.Item>
+
+        <Tooltip
+          title="Password must be atleast 6 characters and alphanumeric!"
```
Comment (Kurotwo) [inline components/RegisterModal.js]: Typo

**Review by Kurotwo — COMMENTED** [review]
Hi @jasonevangelista. We appreciate the effort you put in the project! I'd like to comment on some UI and things I saw.  

1. Password is considered valid despite being only alphabetic.
image

2.  The text for Login and Sign Up in restaurant pages doesn't seem to be vertically centered. I might also suggest making the Login button text white(?) so it contrasts more with the navbar color.
image

3. Typo with the password requirements notif box. 
image

Aside from that, the modal looks great!

**Comment by jasonevangelista** [comment]
Thank you for the suggestions and comments!

@NeilMatthewLua 

- opening modal in landing page now still shows the scroll bar

Screen Shot 2020-12-28 at 3 52 46 PM

@Kurotwo 

- Adjusted header text alignment and colors

Screen Shot 2020-12-28 at 3 52 58 PM

- fixed typo in notif box
- password alphanumeric verification now working properly

Screen Shot 2020-12-28 at 3 53 20 PM

**Comment by NeilMatthewLua** [comment]
@jasonevangelista 

Maybe we can update the hint to tell users na it wont take special char

image

Aside from that, the new updates look great!

**Comment by jasonevangelista** [comment]
> @jasonevangelista
> 
> Maybe we can update the hint to tell users na it wont take special char
> 
> image
> 
> Aside from that, the new updates look great!

hmm something like "Password must be at least 6 characters and must consist of both letters and numbers only" ?

hb yung error message, goods na ba yun na "Password must be alphanumeric!" or change to "Password must  consist of both letters and numbers only"

**Comment by jasonevangelista** [comment]
@NeilMatthewLua hb like this? tho havent pushed this sa repo pa HAHAHA
Screen Shot 2020-12-29 at 3 26 55 PM

**Comment by NeilMatthewLua** [comment]
@jasonevangelista im so sorry your previous comment didn't notify me HUHU tag me nalang if ever HAHAH.  How about "must only be alphanumeric" yung dulo para less lengthy?

**Comment by jasonevangelista** [comment]
@NeilMatthewLua no problem!! HAHAHA like this?

Screen Shot 2020-12-29 at 4 54 05 PM

**Comment by NeilMatthewLua** [comment]
@jasonevangelista looks good, thanks!!

**Review by NeilMatthewLua — APPROVED** [review]

**Pull request merged**